plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java

```diff
@@ -187,14 +187,10 @@ public void allRequiredSinksFinished()
             // input is ready, create exchange source handles
             exchangeSourceHandlesCreationStarted = true;
             exchangeSourceHandlesCreationFuture = stats.getCreateExchangeSourceHandles().record(this::createExchangeSourceHandles);
-            exchangeSourceHandlesFuture.whenComplete((value, failure) -> {
-                if (exchangeSourceHandlesFuture.isCancelled()) {
-                    exchangeSourceHandlesFuture.cancel(true);
-                }
-            });
         }
         if (exchangeSourceHandlesCreationFuture != null) {
-            Futures.addCallback(exchangeSourceHandlesCreationFuture, new FutureCallback<>() {
+            Futures.addCallback(exchangeSourceHandlesCreationFuture, new FutureCallback<>()
+            {
```

Comment:
revert

Reply:
Actually it looks like you can remove the `exchangeSourceHandlesCreationFuture != null` check? Or am I reading it wrong?